codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ServiceGenerator.java

```diff
@@ -223,6 +226,29 @@ private void generateClientInvokeOperation() {
         });
     }
 
+    private void generateSlotInitialization() {
+        // Register stack slots provided by runtime plugins.
+        StackSlotRegistrar slotRegistrar = resolveStackSlotRegistrar();
+        slotRegistrar.generateSlotRegistration(writer, "stack");
+    }
+
+    private StackSlotRegistrar resolveStackSlotRegistrar() {
+        StackSlotRegistrar.Builder builder = StackSlotRegistrar.builder();
+
+        for (RuntimeClientPlugin runtimeClientPlugin : runtimePlugins) {
+            if (!runtimeClientPlugin.matchesService(model, service)
+                    || !runtimeClientPlugin.getResolveFunction().isPresent()) {
+                Optional<StackSlotRegistrar> registrar = runtimeClientPlugin.getRegisterStackSlots();
+                if (!registrar.isPresent()) {
+                    continue;
+                }
+                builder = builder.merge(registrar.get());
+            }
```

Comment:
The `StackSlotRegistrar` builder has a both add and remove slot behavior. How should that get reconciled with the merging, or would it?

Reply:
That was just put there to match the existing builder pattern, but I could see that as being potentially confusing as the removal isn't a process directive, but only removes the entry from the given step if present. Likely we should just remove it from the build. The idea is that there are few plugins that register slots, and likely we shouldn't have a plugin that removes another slot that a plugin might rely on.